pkg/kube/controllers/quarksstatefulset/active_passive_reconciler.go

```diff
@@ -0,0 +1,247 @@
+// loop, the requeue will happen after the ActivePassiveProbe PeriodSeconds is reached.
+func (r *ReconcileStatefulSetActivePassive) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	qSts := &qstsv1a1.QuarksStatefulSet{}
+
+	// Set the ctx to be Background, as the top-level context for incoming requests.
+	ctx, cancel := context.WithTimeout(r.ctx, r.config.CtxTimeOut)
+	defer cancel()
+
+	ctxlog.Info(ctx, "Reconciling for active/passive QuarksStatefulSet", request.NamespacedName)
+
+	if err := r.client.Get(ctx, request.NamespacedName, qSts); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Reconcile successful - don't requeue
+			ctxlog.Infof(ctx, "Failed to find quarks statefulset '%s', not retrying: %s", request.NamespacedName, err)
+			return reconcile.Result{}, nil
+		}
+		// Reconcile failed due to error - requeue
+		ctxlog.Errorf(ctx, "Failed to get quarks statefulset '%s': %s", request.NamespacedName, err)
+		return reconcile.Result{}, err
+	}
+
+	statefulSets, _, err := GetMaxStatefulSetVersion(ctx, r.client, qSts)
+	if err != nil {
+		// Reconcile failed due to error - requeue
+		return reconcile.Result{}, errors.Wrapf(err, "couldn't list StatefulSets for active/passive reconciliation")
+	}
+
+	ownedPods, err := r.getStsPodList(ctx, statefulSets)
+	if err != nil {
+		// Reconcile failed due to error - requeue
+		return reconcile.Result{}, errors.Wrapf(err, "couldn't retrieve pod items from sts: %s", qSts.Name)
+	}
+
+	// retrieves the ActivePassiveProbe children key,
+	// this is the container name in where the ActivePassiveProbe
+	// cmd, needs to be executed
+	containerName, err := getProbeContainerName(qSts.Spec.ActivePassiveProbe)
+	if err != nil {
+		// Reconcile failed due to error - requeue
+		return reconcile.Result{}, errors.Wrapf(err, "None container name found in probe for %s QuarksStatefulSet", qSts.Name)
+	}
+
+	err = r.markActiveContainers(ctx, containerName, ownedPods, qSts)
+	if err != nil {
+		// Reconcile failed due to error - requeue
+		return reconcile.Result{}, err
+	}
+
+	periodSeconds := time.Second * time.Duration(qSts.Spec.ActivePassiveProbe[containerName].PeriodSeconds)
+	if periodSeconds == (time.Second * time.Duration(0)) {
+		ctxlog.WithEvent(qSts, "active-passive").Debugf(ctx, "periodSeconds probe was not specified, going to default to 30 secs")
+		periodSeconds = time.Second * 30
+	}
+
+	// Reconcile for any reason than error after the ActivePassiveProbe PeriodSeconds
+	return reconcile.Result{RequeueAfter: periodSeconds}, nil
+}
+
+func (r *ReconcileStatefulSetActivePassive) markActiveContainers(ctx context.Context, container string, pods *corev1.PodList, qSts *qstsv1a1.QuarksStatefulSet) (err error) {
+
+	probeCmd := qSts.Spec.ActivePassiveProbe[container].Exec.Command
+
+	for _, pod := range pods.Items {
+		ctxlog.WithEvent(qSts, "active-passive").Debugf(
+			ctx,
+			"validating probe in pod: %s",
+			pod.Name,
+		)
+		succeed, _ := r.execContainerCmd(&pod, container, probeCmd)
```

Comment:
You're ignoring err here - I'm assuming err is not nil if there was an api communication failure for example.
We should log this.